### PR TITLE
Set up e2e tests

### PR DESCRIPTION
**Pull request by nandito**
* Add a sample react app that can be used to test the Browser SDK
* Extend `useLocalMedia` to accept `MediaStreams`
* Add some e2e tests using Playwright, run them on GitHub Actions

## Test plan

1. Test the tests: follow the "Running the tests locally" guide in the `test/README.md` doc
2. Test the fake media stream story:
    1. Start storybook: `STORYBOOK_ROOM=<prod-unlocked-room-url> yarn dev`
    2. Open the "Custom UI / Local Media With Fake Media Stream" story
    3. Click the add fake audio track button
    4. Click the connect to room button
    5. Confirm you see the fake webcam and hear the JS-generated noise in your prod room.
3. See the new "Playwright Tests" CI job. It should be green.

## Gotchas

* More tests will be added in upcoming PR(s) for knocking / chat messages / (fake) cam/mic toggles, etc.
* Tests are running on macOS runners, as ubuntu's webkit browser does not fully support webRTC
* Some caching should be added to the CI flow (another upcoming task)

### CODE REVIEW

File: test/sample-app/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
+{
+  "compilerOptions": {
+    "target": "es5",
```
Comment (havardholvik) [inline test/sample-app/tsconfig.json:3]: es5 is very old, Id suggest using es6 here

**Review by havardholvik — APPROVED** [review]
Great addition 🙌 

Just a few suggestions in the code, otherwise this looks like a very sane setup we can easily build upon.

When running the tests locally I got timeout issues when targeting prod, this might be the rate limiting you talked about (used funtimes account). When targeting local-stack, things worked much better, although there was a test failing consistently (see screenshot). The failure appeared both in Firefox and Webkit it seems.

Approving to unblock you, but the failed test should be looked into. Let me know if you need more from me in terms of logs etc.
Screenshot 2023-10-11 at 21 54 42

File: src/stories/custom-ui.stories.tsx [hunk 2]
```diff
@@ -14,6 +16,23 @@ export default {
 
 const roomRegEx = new RegExp(/^https:\/\/.*\/.*/);
 
+function fakeAudioStream() {
```
[the remark this answers is not in the document]
Comment (nandito) [reply src/stories/custom-ui.stories.tsx:19]: Yeah I was thinking about the same, but as this was a short one I added it here directly. 🙈 But good shout, I put it to lib/utils 👍

**Comment by nandito** [comment]
I can't reproduce your test failures locally. 😕 @thyal tested it too and they passed on his device and the CI also seems to be green.

Have you installed the test browsers?

**Comment by thyal** [comment]
confirming all tests passed for me locally 👍

**Pull request merged**